apps/myjobs/app/models/workflow_file.rb

```diff
@@ -0,0 +1,47 @@
+class WorkflowFile
+  attr_reader :path, :staged_dir
+
+  def initialize(path, staged_dir)
+    @path = path
+    @staged_dir = staged_dir
+  end
+
+  # Get an array of files of a directory meeting the criteria for job scripts in alphabetical order
+  #
+  # Filter the files by file size, extension and content
+  def suggested_script?
+      valid_size? && (has_suggested_extensions? || starts_with_shebang_line? || has_specific_command?)
+  end
+ 
+  # Return true if the file meets the basic requirements to be a job script
+  def valid_script?
+    @valid_script ||= valid_size?
+  end
+  
+  # Return true if file name extention is in the recommended extension list
+  def has_suggested_extensions?
+    suggested_job_script_file_extensions = [".sh", ".job", ".slurm", ".batch", ".qsub", ".sbatch", ".srun", ".bsub"]
+    suggested_job_script_file_extensions.include? File.extname(path)
+  end
+  
+  # Return true if file starts with a shebang line
+  def starts_with_shebang_line?
+    (File.open(path) { |f| f.read(2) }) == "#!"
```

Comment:
Test if a file has permissions 000 or something funny like that and make sure this doesn't break.

Reply:
Good call! It does raise file permission errors. I added rescue to the function.